### ISSUE TITLE
ASS subs are not rendered correctly in gpu-next

### ISSUE THREAD

**Issue by Kapppa** [mpv-player/mpv#14442]
### mpv Information

```bash
mpv v0.38.0-525-g265056fa Copyright © 2000-2024 mpv/MPlayer/mplayer2 projects
 built on Jun 23 2024 00:07:06
libplacebo version: v7.349.0 (v7.349.0-rc1-3-g1fd3c7b-dirty)
FFmpeg version: N-115975-g0c0e7ec81
FFmpeg library versions:
   libavcodec      61.8.100
   libavdevice     61.2.100
   libavfilter     10.2.102
   libavformat     61.3.104
   libavutil       59.25.100
   libswresample   5.2.100
   libswscale      8.2.100
```

### Other Information

- Windows version: Microsoft Windows [Version 10.0.19045.4291]
- GPU model, driver and version: RTX 3080, 551.86
- Source of mpv: shinchiro latest windows
- Introduced in version: Unknown

### Reproduction Steps

Load provided video with gpu-next. `--no-config --vo=gpu-next`

### Expected Behavior

Subs are displayed like in normal rending. 
![expected](https://github.com/mpv-player/mpv/assets/17926797/cd17c366-7738-45b9-81fb-330606590bb8)

### Actual Behavior

Subs are displayed incorrectly, looks like some frames are dropped or the layers are displayed in the wrong order. Instead of being white with a purple border they are just purple. 
![actual](https://github.com/mpv-player/mpv/assets/17926797/3a416168-b9b6-4bbe-b859-511e9970911c)

### Log File

[Expected](https://github.com/user-attachments/files/15985082/output.txt)
[Actual](https://github.com/user-attachments/files/15984941/output.txt)

### Sample Files

[issue.zip](https://github.com/user-attachments/files/15984983/issue.zip)

### I carefully read all instruction and confirm that I did the following:

- [X] I tested with the latest mpv version to validate that the issue is not already fixed.
- [X] I provided all required information including system and mpv version.
- [X] I produced the log file with the exact same set of files, parameters, and conditions used in "Reproduction Steps", with the addition of `--log-file=output.txt`.
- [X] I produced the log file while the behaviors described in "Actual Behavior" were actively observed.
- [X] I attached the full, untruncated log file.
- [X] I attached the backtrace in the case of a crash.

**Comment by llyyr** [comment]
It's not that the color is incorrect, the subtitles aren't rendered at all. Only the purple drop shadow is rendered. You can get it to render at arbitrary zoom and panscan though, which is odd

**Comment by Kapppa** [comment]
> It's not that the color is incorrect, the subtitles aren't rendered at all. Only the purple drop shadow is rendered. You can get it to render at arbitrary zoom and panscan though, which is odd
> 
> edit:
> 
> There's a mask(?) to hide [this part](https://github.com/mpv-player/mpv/assets/6298234/3e4a911f-bd07-4b6a-abd5-48609f26379c) of the original video, when enough of the mask is off screen, the subtitle is rendered correctly.

Yeah I shouldn't said frames I meant layers. I assume the GPU rendering is dropping certain layers in the ass file.

**Comment by kasper93** [comment]
> Due to the way subtitles are rendered in mpv, only blend-subtitles=video (which is how VSFilter renders subtitles but isn't supported in gpu-next) can render this correctly.

It is supported in a way that it would be rendered relative to the video, but blended after scaling, which is in theory fine, if the scaling is done correctly. That said, it is likely fixable, even without blending at video size. Thoughts?

**Comment by kasper93** [comment]
> isn't 100% correct because it fails when video and subtitle resolutions don't match

This is the case where it should render subtitles to proper rect. It doesn't mean we have to blend before, it means that we shouldn't use video rect directly.

**Comment by kasper93** [comment]
Will be fixed by https://code.videolan.org/videolan/libplacebo/-/merge_requests/719 That was fun one, took me longer than expected to notice what was going on.
